[luci] Add a wrong type target test to QuantizedModelVerifier test

dayo09 opened the pull request:
This commit adds a wrong type test function with a specified target node.

ONE-DCO-1.0-Signed-off-by: Dayoung Lee <dayg502@gmail.com>

dayo09 commented on compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:
```diff
@@ -1144,7 +1172,23 @@ TEST(QuantizedModelVerifierTest, ArgMax_wrong_dimension_type_NEG)
 }
 
-// TODO : Add a negative test which sets wrong dtype on dimension node
+TEST(QuantizedModelVerifierTest, ArgMax_wrong_dimension_type_NEG)
```
This test code ensures that the above test function works well.

dayo09 commented on compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:
```diff
@@ -114,6 +114,7 @@ void quantize_and_verify(loco::Graph *g, Type quantized_dtype, Granularity granu
 
 // Helper function to reduce duplicate test codes
 // Assumption: g->output()->from() is the target node
+// TODO: Remove the function and use below function to explicit target node
```
The previous helper functions assumed the target node as g->output->from() node, but now that we have both speficied-target test helper, we can replace the function to explicit the target node.

dayo09 replied:
@jinevening PTAL :-D

jinevening replied:
IMHO, I think we can keep this function to avoid changing test cases.

Instead, we can implement this function with your function (with the `target`) to reduce duplicate codes.

seanshpark commented:
@dayo09 , there are mutliple purpose of changes and this makes me hard to understand the code. please split the changes as short as possible you can.
and it would be better not to use explanations like `with a specified target node.` that I dont have any idea what this `specific` means.

seanshpark commented:
This PR is about `output->from()` to `dimension()` of ArgMax if I understand the changes.
Can you explain why this change is needed?

dayo09 commented:
@seanshpark Previously it was dimension() test. But as I refactor argmax test code https://github.com/Samsung/ONE/pull/6637/files, it was impossible to keep so I have replaced them with TODO command.

```
(line 1147) // TODO : Add a negative test which sets wrong dtype on dimension node
```

Now, this code implements that TODO.

seanshpark commented:
> Previously it was dimension() test. But as I refactor argmax test code

It's a bit hard to know why we need to change from `out->from()` to `dimension()`.
As I understand #6637 was a change to previous one kind of test to using macro for multple kinds of test.
Do we need to roleback it? Or are you planning to add test for `out->from()` too?

seanshpark commented:
>  it was impossible to keep

what was impossible to keep? and why? I think we need to solve this problem.
why I am asking is that I don't fully understand you changes and @jinevening is absent to look the codes with domain knowledge.

dayo09 commented:
@seanshpark 
It has already merged. Let me repeat the commit message in the pr.

> Specifically, it tests with wrong type not with dimension but with input node, with more test variant.
...
> The removed wrong type test with dimension node will be revived after the implementation of proper test macros.
```

1. At first, @jinevening made a test with `dimension` node
2. At #6637, I have removed the test and replaced with TODO comment.
   And I added the test with `input` node but mistakenly used the test name `ArgMax_wrong_dimension_type_NEG`
3. At #6655, I have fixed the mistaken name.
4. At this draft, it revives `dimension` node test.

seanshpark commented:
@dayo09 , thanks for the explanation. Please explain next time prior to doing some fixes to reduce confusion.

dayo09 commented:
Rebased :-D

jinevening reviewed (approved):
LGTM

dayo09 commented:
All the prs from this draft are merged. :-D
